packages/space-opera/src/components/mobile_view/mobile_view.ts

```diff
@@ -138,32 +142,67 @@ export class MobileView extends ConnectedLitElement {
   }
 
   async fetchLoop() {
-    await fetch(this.updatesPipeUrl)
-        .then(response => response.json())
-        .then(json => this.waitForData(json))
-        .catch((error) => {
-          console.error('Error:', error);
-        });
+    const response = await fetch(this.updatesPipeUrl);
+    if (response.ok) {
+      const json = await response.json();
+      this.waitForData(json)
+    } else {
+      console.error('Error:', response);
+    }
   }
 
   async triggerFetchLoop() {
     await this.fetchLoop();
     await this.triggerFetchLoop();
   }
 
+  async prepareGlbBlob(gltf: GltfModel) {
+    const glbBuffer = await gltf.packGlb();
+    return new Blob([glbBuffer], {type: 'model/gltf-binary'});
+  }
+
+  // Post new blob for scene-viewer
+  async postSceneViewerBlob(blob: Blob) {
+    const response = await fetch(this.modelViewerUrl, {
+      method: 'POST',
+      body: blob,
+    })
+    if (response.ok) {
+      console.log('Success:', response);
+    }
+    else {
+      throw new Error(`Failed to post: ${this.modelViewerUrl}`);
+    }
+  }
+
+  async newModel() {
```

Comment:
when a new model is loaded, we need to prepare it to be posted again, and do an initial post for sceneviewer